### PR TITLE
V2 : support for colors in dxf serializer

### LINKED ISSUE #640: DXF: create DXF entities with colors
## Expected Behavior

Geometries with colors are exported to DXF with a color attribute.

## Actual Behavior

No colors.

## Steps to Reproduce the Problem

  1. Create a design with a simple shape
  2. Assign a color to the shape
  3. Convert the geometry to DXF

## Specifications

  - Version: V1 V2
  - Platform: All
  - Environment: All

### PR DESCRIPTION
**Pull request by z3dev**
These changes add support for colors when exporting / serializing to the DXF format.

There were some fixes to the DXF deserializer as well, correcting color lookups.

Fixes #640 

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

Testing of colors was also performed by using Autodesk Viewer (on-line viewer by Autocad)

### CODE REVIEW

**Comment by SimonClark** [comment]
sorry, will get on this soon

**Comment by z3dev** [comment]
> sorry, will get on this soon

Thanks.

Don't forget #672 is pending, and awaiting your review as well.

File: packages/io/dxf-serializer/index.js [hunk 1]
```diff
@@ -405,6 +431,34 @@ const getEntityId = (options) => {
...
 }
 
+const getName = (object, options) => {
+  if (object.name) return object.name
+  // add more zeros if the id needs to be larger
+  const padded = '00000' + options.entityId.toString(16).toUpperCase()
+  return 'CAD' + padded.substr(padded.length - 5)
+}
+
+const getColorNumber = (object, options) => {
```
Comment (SimonClark) [inline packages/io/dxf-serializer/index.js:441]: DXF color table seems to be organized into the most used colors first. It's probably worth exiting the loop if diff == 0, as most calls will then not need to iterate over all 256 entries.

**Review by SimonClark — APPROVED** [review]
Minor optimization suggested. Take it or leave it.  Otherwise +1.

File: packages/io/dxf-serializer/index.js (hunk 1 above)
Comment (z3dev) [reply packages/io/dxf-serializer/index.js:441]: Nice suggestion. I’ll make a small change..

**Review by SimonClark — APPROVED** [review]
👍

**Pull request merged**